### ISSUE TITLE
memory and priority idea

### ISSUE THREAD

**Issue by Alessandro-Barbieri** [BOINC/boinc#1355]
See http://lists.ssl.berkeley.edu/pipermail/boinc_dev/2015-February/021622.html

**Comment by ChristianBeer** [comment]
Email archive no longer reachable. If someone knows what this refers to please reopen with a proper link.

**Comment by Melebius** [comment]
The link target is still available in the [Google cache](http://webcache.googleusercontent.com/search?q=cache:mKRfLSLcd6cJ:lists.ssl.berkeley.edu/pipermail/boinc_dev/2015-February/021622.html). I am copying it here because Google probably won’t keep it forever. However, I would need to perform more investigation to prove I am really affected by this issue, so I am keeping it closed for now.

```
Il mer 4 feb 2015 00:44 CET, Alessandro Barbieri ha scritto:

Package: boinc-client
Version: 7.4.23+dfsg-1
Severity: wishlist
Tags: upstream

Dear Maintainer,

when a WU is waiting for memory can the manager run a lower priority WU (that uses less memory)
instead of waiting and not using all cores?

-- Package-specific info:
-- Contents of /etc/default/boinc-client:
# This file is /etc/default/boinc-client, it is a configuration file for the
# /etc/init.d/boinc-client init script.

# Set this to 1 to enable and to 0 to disable the init script.
ENABLED="1"

# Set this to 1 to enable advanced scheduling of the BOINC core client and
# all its sub-processes (reduces the impact of BOINC on the system's
# performance).
SCHEDULE="1"

# The BOINC core client will be started with the permissions of this user.
BOINC_USER="boinc"

# This is the data directory of the BOINC core client.
BOINC_DIR="/var/lib/boinc-client"

# This is the location of the BOINC core client, that the init script uses.
# If you do not want to use the client program provided by the boinc-client
# package, you can specify here an alternative client program.
#BOINC_CLIENT="/usr/local/bin/boinc"
BOINC_CLIENT="/usr/bin/boinc"

# Here you can specify additional options to pass to the BOINC core client.
# Type 'boinc --help' or 'man boinc' for a full summary of allowed options.
#BOINC_OPTS="--allow_remote_gui_rpc"
BOINC_OPTS="

# Scheduling options

# Set SCHEDULE="0" if prefering to run with upstream default priority
# settings.

# Nice levels. When systems are truly busy, e.g. because of too many active
# scientific applications started by the boinc client, there is a chance for
# the boinc client not to be granted sufficient opportunity to check for
# scientific applications to be alive and make the (wrong) decision to
# terminate the scientific app. This is particularly an issue with many
# apps started in parallel on modern multi-core systems and extra overheads
# for the download and uploads of files with the project servers. Another
# concern is the latency for scientific applications to communicate with the
# graphics card, which should be low. All such values should be set and
# controled from within the BOINC client. The Debian init script also sets
# extra constrains via chrt on real time performance and via ionice on
# I/O performance, which is beyond the regular BOINC client. It then was
# too easy to use that code to also constrain minimal nice levels. We still
# think about how to best distinguish GPU applications from regular apps.
BOINC_NICE_CLIENT=10
BOINC_NICE_APP_DEFAULT=19
#BOINC_NICE_APP_GPU=5        # not yet used

# ionice classes. See manpage of ionice (1) in the util-linux package.
BOINC_IONICE_CLIENT=3        # idle
#BOINC_IONICE_APP_DEFAULT=3  # idle, not yet used
#BOINC_IONICE_APP_GPU=2      # best effort, not yet used


-- System Information:
Debian Release: 8.0
  APT prefers testing
  APT policy: (650, 'testing'), (500, 'testing-updates'), (500, 'testing-proposed-updates'), (1, 'experimental')
Architecture: amd64 (x86_64)
Foreign Architectures: i386

Kernel: Linux 3.16.0-4-amd64 (SMP w/4 CPU cores)
Locale: LANG=it_IT.UTF-8, LC_CTYPE=it_IT.UTF-8 (charmap=UTF-8)
Shell: /bin/sh linked to /bin/dash
Init: systemd (via /run/systemd/system)

Versions of packages boinc-client depends on:
ii  adduser                3.113+nmu3
ii  ca-certificates        20141019
ii  debconf [debconf-2.0]  1.5.55
ii  libboinc7              7.4.23+dfsg-1
ii  libc6                  2.19-13
ii  libcurl3               7.38.0-4
ii  libgcc1                1:4.9.1-19
ii  libstdc++6             4.9.1-19
ii  python                 2.7.8-2
ii  zlib1g                 1:1.2.8.dfsg-2+b1

boinc-client recommends no packages.

Versions of packages boinc-client suggests:
pn  boinc-amd-opencl   <none>
ii  boinc-manager      7.4.23+dfsg-2exp1
ii  boinc-nvidia-cuda  7.4.23+dfsg-2exp1
ii  x11-xserver-utils  7.7+3+b1

-- Configuration Files:
/etc/boinc-client/cc_config.xml changed:
<cc_config>
    <log_flags>
        <file_xfer>1</file_xfer>
        <sched_ops>1</sched_ops>
        <task>1</task>
        <android_debug>0</android_debug>
        <app_msg_receive>0</app_msg_receive>
        <app_msg_send>0</app_msg_send>
        <async_file_debug>0</async_file_debug>
        <benchmark_debug>0</benchmark_debug>
        <checkpoint_debug>0</checkpoint_debug>
        <coproc_debug>0</coproc_debug>
        <cpu_sched>0</cpu_sched>
        <cpu_sched_debug>0</cpu_sched_debug>
        <cpu_sched_status>0</cpu_sched_status>
        <dcf_debug>0</dcf_debug>
        <disk_usage_debug>0</disk_usage_debug>
        <file_xfer_debug>0</file_xfer_debug>
        <gui_rpc_debug>0</gui_rpc_debug>
        <heartbeat_debug>0</heartbeat_debug>
        <http_debug>0</http_debug>
        <http_xfer_debug>0</http_xfer_debug>
        <mem_usage_debug>0</mem_usage_debug>
        <network_status_debug>0</network_status_debug>
        <notice_debug>0</notice_debug>
        <poll_debug>0</poll_debug>
        <priority_debug>0</priority_debug>
        <proxy_debug>0</proxy_debug>
        <rr_simulation>0</rr_simulation>
        <rrsim_detail>0</rrsim_detail>
        <sched_op_debug>0</sched_op_debug>
        <scrsave_debug>0</scrsave_debug>
        <slot_debug>0</slot_debug>
        <state_debug>0</state_debug>
        <statefile_debug>0</statefile_debug>
        <suspend_debug>0</suspend_debug>
        <task_debug>0</task_debug>
        <time_debug>0</time_debug>
        <trickle_debug>0</trickle_debug>
        <unparsed_xml>0</unparsed_xml>
        <work_fetch_debug>0</work_fetch_debug>
    </log_flags>
    <options>
        <abort_jobs_on_exit>0</abort_jobs_on_exit>
        <allow_multiple_clients>0</allow_multiple_clients>
        <allow_remote_gui_rpc>0</allow_remote_gui_rpc>
        <client_version_check_url>http://boinc.berkeley.edu/download.php?xml=1</client_version_check_url>
        <client_new_version_text></client_new_version_text>
        <client_download_url>http://boinc.berkeley.edu/download.php</client_download_url>
        <disallow_attach>0</disallow_attach>
        <dont_check_file_sizes>0</dont_check_file_sizes>
        <dont_contact_ref_site>0</dont_contact_ref_site>
        <exclusive_app>iceweasel</exclusive_app>
        <exclusive_app>josm</exclusive_app>
        <exclusive_app>LolClient.exe</exclusive_app>
        <exclusive_app>mplayer2</exclusive_app>
        <exclusive_app>SC2.exe</exclusive_app>
        <exclusive_app>StarCraft II.exe</exclusive_app>
        <exclusive_app>virtualbox</exclusive_app>
        <exit_after_finish>0</exit_after_finish>
        <exit_before_start>0</exit_before_start>
        <exit_when_idle>0</exit_when_idle>
        <fetch_minimal_work>0</fetch_minimal_work>
        <fetch_on_update>0</fetch_on_update>
        <force_auth>default</force_auth>
        <http_1_0>0</http_1_0>
        <http_transfer_timeout>300</http_transfer_timeout>
        <http_transfer_timeout_bps>10</http_transfer_timeout_bps>
        <max_event_log_lines>2000</max_event_log_lines>
        <max_file_xfers>8</max_file_xfers>
        <max_file_xfers_per_project>2</max_file_xfers_per_project>
        <max_stderr_file_size>0</max_stderr_file_size>
        <max_stdout_file_size>0</max_stdout_file_size>
        <max_tasks_reported>0</max_tasks_reported>
        <ncpus>-1</ncpus>
        <network_test_url>http://www.google.com/</network_test_url>
        <no_alt_platform>0</no_alt_platform>
        <no_gpus>0</no_gpus>
        <no_info_fetch>0</no_info_fetch>
        <no_priority_change>0</no_priority_change>
        <os_random_only>0</os_random_only>
<proxy_info>
    <socks_server_name></socks_server_name>
    <socks_server_port>80</socks_server_port>
    <http_server_name></http_server_name>
    <http_server_port>80</http_server_port>
    <socks5_user_name></socks5_user_name>
    <socks5_user_passwd></socks5_user_passwd>
    <http_user_name></http_user_name>
    <http_user_passwd></http_user_passwd>
    <no_proxy></no_proxy>
</proxy_info>
        <rec_half_life_days>10.000000</rec_half_life_days>
        <report_results_immediately>0</report_results_immediately>
        <run_apps_manually>0</run_apps_manually>
        <save_stats_days>30</save_stats_days>
        <skip_cpu_benchmarks>0</skip_cpu_benchmarks>
        <simple_gui_only>0</simple_gui_only>
        <start_delay>0.000000</start_delay>
        <stderr_head>0</stderr_head>
        <suppress_net_info>0</suppress_net_info>
        <unsigned_apps_ok>0</unsigned_apps_ok>
        <use_all_gpus>0</use_all_gpus>
        <use_certs>0</use_certs>
        <use_certs_only>0</use_certs_only>
        <vbox_window>0</vbox_window>
    </options>
</cc_config>

/etc/boinc-client/global_prefs_override.xml changed:
<global_preferences>
   <run_on_batteries>0</run_on_batteries>
   <run_if_user_active>1</run_if_user_active>
   <run_gpu_if_user_active>0</run_gpu_if_user_active>
   <idle_time_to_run>3.000000</idle_time_to_run>
   <suspend_cpu_usage>20.000000</suspend_cpu_usage>
   <start_hour>0.000000</start_hour>
   <end_hour>0.000000</end_hour>
   <net_start_hour>0.000000</net_start_hour>
   <net_end_hour>0.000000</net_end_hour>
   <leave_apps_in_memory>0</leave_apps_in_memory>
   <confirm_before_connecting>0</confirm_before_connecting>
   <hangup_if_dialed>0</hangup_if_dialed>
   <dont_verify_images>0</dont_verify_images>
   <work_buf_min_days>0.010000</work_buf_min_days>
   <work_buf_additional_days>0.500000</work_buf_additional_days>
   <max_ncpus_pct>100.000000</max_ncpus_pct>
   <cpu_scheduling_period_minutes>120.000000</cpu_scheduling_period_minutes>
   <disk_interval>1.000000</disk_interval>
   <disk_max_used_gb>0.000000</disk_max_used_gb>
   <disk_max_used_pct>50.000000</disk_max_used_pct>
   <disk_min_free_gb>0.500000</disk_min_free_gb>
   <vm_max_used_pct>90.000000</vm_max_used_pct>
   <ram_max_used_busy_pct>50.000000</ram_max_used_busy_pct>
   <ram_max_used_idle_pct>75.000000</ram_max_used_idle_pct>
   <max_bytes_sec_up>10240.000000</max_bytes_sec_up>
   <max_bytes_sec_down>20480.000000</max_bytes_sec_down>
   <cpu_usage_limit>100.000000</cpu_usage_limit>
   <daily_xfer_limit_mb>0.000000</daily_xfer_limit_mb>
   <daily_xfer_period_days>0</daily_xfer_period_days>
</global_preferences>

/etc/boinc-client/gui_rpc_auth.cfg [Errno 13] Permesso negato: u'/etc/boinc-client/gui_rpc_auth.cfg'

-- debconf information:
  boinc-client/remove_boinc_dir: true
```

**Comment by Ageless93** [comment]
The link to Pipermail works for me. Perhaps that at the time you checked,
the mail archive was on one of the servers that were out of the air due to
maintenance @Berkeley.


-- Jord van der Elst.

**Comment by LocutusOfBorg** [comment]
Can you please reopen it?
